shm/island.cpp

```diff
@@ -0,0 +1,8 @@
+#include "island.hpp"
+#include "coordinates.hpp"
+
+Island::Island(int positionX, int positionY) : position_(Coordinates(positionX, positionY)) {}
```

Comment:
this constructor is strange. Class Island should be initialized with coordinates instead of the elements of hte coordinates. so I would expect something like this: Island::Island(Coordinates position) : position_(position) {}

Reply:
I changed the constructor to use Coordinates as a parameter. Now we have to remember that we have to create Coordinates first. Mateusz suggested somewhere to use positionX and positionY as a parameters, but I agree with you, it will be easier to understand